src/models/Package.vala

```diff
@@ -0,0 +1,165 @@
+                }
+
+                if (property_node.get_value_type () != Type.STRING) {
+                    return false;
+                }
+
+                var regex = /^(.*)(?:\s)(<.*>)(?:\s)(\(.*\))/;
+                MatchInfo info;
+                
+                if (regex.match (property_node.get_string (), 0, out info)) {
+                    @value = new Person () {
+                        name = info.fetch (1),
+                        email = info.fetch (2),
+                        url = info.fetch (3),
+                    };
+
+                    return true;
+                }
+
+                return false;
+
+            default:
+                return default_deserialize_property (
+                    property_name,
+                    out @value,
+                    pspec,
+                    property_node
+                );
+        }
+    }
+}
+
+class Vamp.Repository : Object {
+    public string type { get; set; }
+    public string url { get; set; }
+    public string directory { get; set; }
+
+    public static Repository from_json (Json.Node node) {
+        assert (node.get_node_type () == OBJECT);
+        return (Repository) Json.gobject_deserialize (typeof (Repository), node);
+    }
+
+    public Json.Node to_json () {
+        return Json.gobject_serialize (this);
+    }
+}
+
+class Vamp.FundingInfo : Object {
+    public string type { get; set; }
+    public string url { get; set; }
+
+    public static FundingInfo from_json (Json.Node node) {
+        assert (node.get_node_type () == OBJECT);
+        return (FundingInfo) Json.gobject_deserialize (typeof (FundingInfo), node);
+    }
+
+    public Json.Node to_json () {
+        return Json.gobject_serialize (this);
+    }
+}
+
+class Vamp.Person : Object {
+    public string name { get; set; }
+    public string email { get; set; }
+    public string url { get; set; }
+
+    public static Person from_json (Json.Node node) {
+        assert (node.get_node_type () == OBJECT);
+        return (Person) Json.gobject_deserialize (typeof (Person), node);
+    }
+
+    public Json.Node to_json () {
+        return Json.gobject_serialize (this);
+    }
+}
+
+class Vamp.Bugs : Object {
+    public string url { get; set; }
+    public string email { get; set; }
+
+    public static Bugs from_json (Json.Node node) {
+        assert (node.get_node_type () == OBJECT);
+        return (Bugs) Json.gobject_deserialize (typeof (Bugs), node);
+    }
+
+    public Json.Node to_json () {
+        return Json.gobject_serialize (this);
+    }
+}
+
+Gee.List<string> string_list_from_json (Json.Node node) {
+    assert (node.get_node_type () == ARRAY);
+
+    var array = node.get_array ();
+    var result = new Gee.ArrayList<string> ();
+
+    array.foreach_element ((_, __, element_node) => {
+        if (element_node.get_value_type () == Type.STRING) {
+            return;
+        }
+
+        result.add(element_node.get_string ());
+    });
+
+    return result;
+}
```

Comment:
I think we should have one file per public class: https://docs.elementary.io/develop/writing-apps/code-style#classes-and-files

Also: we will have this classes on the `Vamp` namespace or in a more specific namespace?